alf/algorithms/sarsa_algorithm.py

```diff
@@ -196,13 +203,19 @@ def __init__(self,
         self._use_entropy_reward = False
         if initial_alpha is not None:
             if isinstance(actor_network, DistributionNetwork):
-                if target_entropy is None:
+                if target_entropy is None or callable(target_entropy):
+                    if target_entropy is None:
+                        target_entropy = dist_utils.calc_default_target_entropy
                     target_entropy = np.sum(
-                        list(
-                            map(dist_utils.calc_default_target_entropy,
-                                flat_action_spec)))
+                        list(map(target_entropy, flat_action_spec)))
+                    logging.info(
+                        "Target entropy is calculated for {}: {}.".format(
+                            self.name, target_entropy))
+                else:
+                    logging.info(
+                        "User-supplied target entropy for {}: {}".format(
+                            self.name, target_entropy))
```

Comment:
This part seems to be shared with SAC and potentially other future algorithms with target entropy. It might be better to put them into a single shared function?